### PR TITLE
Add remembered state for smart annotation [SCI-4950]

### PR DESCRIPTION
**Pull request by aignatov-bio**
Jira ticket: [SCI-4950](https://biosistemika.atlassian.net/browse/SCI-4950)

### What was done
Add remembered state for smart annotation

### CODE REVIEW

File: app/assets/javascripts/sitewide/atwho_res.js.erb [hunk 1]
```diff
@@ -103,6 +103,11 @@ var SmartAnnotation = (function() {
 
             }
             $.getJSON(filterType.dataUrl, params, function(data) {
+
+              localStorage.setItem('scinote-smart-annotation-state-per-team-' + data.team, JSON.stringify({
```
Comment (okriuchykhin) [inline app/assets/javascripts/sitewide/atwho_res.js.erb:107]: I think there is no need in 'scinote' prefix.
And I think we use '_' elsewhere in key names

File: app/assets/javascripts/sitewide/atwho_res.js.erb [hunk 2]
```diff
@@ -162,8 +169,18 @@ var SmartAnnotation = (function() {
         })
 
         if ($currentAtWho.find('.tab-pane.active').length == 0) {
-          let filterType =  DEFAULT_SEARCH_FILTER;
-          $currentAtWho.find(`.${filterType.tag}`).click();
+          let filterType =  DEFAULT_SEARCH_FILTER.tag;
+          let teamId = $currentAtWho.find('.atwho-header-res').data('team-id');
+          let remeberedState = localStorage.getItem('scinote-smart-annotation-state-per-team-' + teamId);
```
Comment (okriuchykhin) [inline app/assets/javascripts/sitewide/atwho_res.js.erb:174]: Maybe better 'smart_annotation_states/teams/{id}'

File: app/assets/javascripts/sitewide/atwho_res.js.erb [hunk 3]
```diff
@@ -162,8 +169,18 @@ var SmartAnnotation = (function() {
...
+          if (remeberedState) {
+            try {
+              remeberedState = JSON.parse(remeberedState);
+              filterType = remeberedState.tag;
+              $currentAtWho.find(`.repository-object[data-object-id=${remeberedState.repository}]`)
+                .addClass('btn-primary');
+            } catch {}
```
Comment (okriuchykhin) [inline app/assets/javascripts/sitewide/atwho_res.js.erb:181]: shouldn't it reset state if it is corrupted?

Comment (aignatov-bio) [reply app/assets/javascripts/sitewide/atwho_res.js.erb:181]: filterType initially has a default state `let filterType =  DEFAULT_SEARCH_FILTER.tag;`

**Review by okriuchykhin — APPROVED** [review]

**Pull request merged**